Procurement/ViewModel/Recipes/VaalOrbRecipe.cs

```diff
@@ -0,0 +1,67 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using POEApi.Model;
+
+namespace Procurement.ViewModel.Recipes
+{
+    internal class VaalOrbRecipe : Recipe
+    {
+        public override string Name => "1 Vaal Orb";
+
+        public override IEnumerable<RecipeResult> Matches(IEnumerable<Item> items)
+        {
+            List<RecipeResult> recipeSets = new List<RecipeResult>();
+
+            var candidateGems = items.OfType<Gem>().Where(gem => gem.Corrupted).Cast<Item>().ToList();
+
+            var fragments = items.Where(x => x.TypeLine.StartsWith("Sacrifice at", StringComparison.CurrentCultureIgnoreCase)).ToList();
+            while(candidateGems.Count > 0)
+            {
+                var recipeResult = new RecipeResult()
+                {
+                    Instance = this,
+                    MatchedItems = new List<Item>(),
+                    Missing = new List<string>(),
+                    IsMatch = true
+                };
+
+                for (int i = candidateGems.Count - 1; i >= 0; i--)
+                {
+                    var candidateGem = candidateGems[i];
+                    if (recipeResult.MatchedItems.Count <= 6)
```

Comment:
Instead of using magic numbers `6` and `7`, create a class member representing the required number of gems, and use that where required.

Reply:
Added constant and changed logic so that only 7 is needed.